src/logger/logger-wrapper.ts

```diff
@@ -40,24 +39,20 @@ function isInitialized(): boolean {
  *
  * @returns { IVSCodeExtLogger }
  */
-export function getLogger(): IVSCodeExtLogger {
+export function getLogger(): IVSCodeExtLogger | undefined {
```

Comment:
Without it I get compilation error: Type 'undefined' is not assignable to type 'IVSCodeExtLogger'

Reply:
Because this is how `logger` is defined. But you check just before it that it's not undefined.
If you want typescript to automatically recognize that it's not undefined you can do a small refactor - send `logger` to `isInitialized` and can change the return type of `isInitialized` to a type assertion (`logger is IVSCodeExtLogger`).
But it's not really necessary, you can just return `logger!` and suppress the `@typescript-eslint/no-non-null-assertion` rule for that line (with a relevant comment of course - that `isInitialized` already checks it)

Reply:
Done